azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -232,7 +244,7 @@ def __init__(self, kcsb):
         kusto_cluster = kcsb.data_source
         self._mgmt_endpoint = "{0}/v1/rest/mgmt".format(kusto_cluster)
         self._query_endpoint = "{0}/v2/rest/query".format(kusto_cluster)
-        self._aad_helper = _AadHelper(kcsb)
+        self._aad_helper = _AadHelper(kcsb) if kcsb.aad_federated_security else None

```

Comment:
following previous comment, this should be reversed. if federated, then no point of using aad.